### DISCUSSION TITLE
React Native Web: Dynamic Page Rendering?

### DISCUSSION THREAD

**Discussion by k2on** [expo/expo#28011] [Archived: Questions]
With Expo 50, is it now possible to do SSR with dynamic data, so I can include dynamic metadata in the html (opengraph)?

**Comment by FleetAdmiralJakob** [comment]
Hey, could you please share the blog post, the announcement, or the docs for SSR with Expo for Web?

**Reply by k2on** [reply]
As of now SSR (at request time) is not really supported.

https://docs.expo.dev/router/reference/static-rendering/#server-side-rendering

But...

You can use api routes to render HTML to hack it together.

Here is a simple [demo](https://github.com/k2on/rn-ssr)

But it should be supported when [this](https://www.youtube.com/watch?v=BK2xbPW2uUU) magic gets released.

**Reply by FleetAdmiralJakob** [reply]
Crazy. Server Components with Expo will be so mind-blowing. Why should you even build a Next.js app if you get everything with Expo and native support on top.
